### PR TITLE
Datadog tracing options

### PR DESCRIPTION
**Pull request by stroem**

### CODE REVIEW

File: client/options.go [hunk 1]
```diff
@@ -41,16 +41,16 @@ func WithOpenCensusTracing() Option {
 // so that it will automatically inject trace-headers.
 //
 // Should be used when you trace your application with Datadog.
-func WithDatadogTracing(serviceName string) Option {
+func WithDatadogTracing(opts ...dd_http.RoundTripperOption) Option {
```
Comment (Ekberg) [inline client/options.go:44]: Is serviceName optional now? If not maybe keep the serviceName argument and just add the ability to append other options?

Comment (stroem) [reply client/options.go:44]: There is multiple ways to set a service name, like for instance starting the tracer:
```
ddtracer.Start(ddtrace.WithService("my-service"))
...
```

This solution will make it possible to both set the service name by options in this function, using starting the tracer or other ways

Comment (lonnblad) [reply client/options.go:44]: in the case of lambda, you can't set it on the tracer, nor the agent, so we need to set it through options, I believe it needs to be clear somewhere that to use this with a lambda you need to add the service-name option

File: client/options.go [hunk 2]
```diff
@@ -41,16 +41,16 @@ func WithOpenCensusTracing() Option {
...
 	resourceNamer := func(req *http.Request) string {
 		return fmt.Sprintf("%s %s", req.Method, req.URL.String())
 	}
+	opts = append(opts, dd_http.RTWithResourceNamer(resourceNamer))
```
Comment (lonnblad) [inline client/options.go:48]: should this check if there is already an ResourceNamer option?

File: client/options.go (hunk 1 above)
Comment (stroem) [reply client/options.go:44]: @lonnblad Any suggestions how to make this clear?

File: client/options.go (hunk 2 above)
Comment (stroem) [reply client/options.go:48]: Thats not possible if we dont make our own options (which I prefer not to do).
I will update the PR to make it possible to set your own resourcenamer:
```
opts = append([]dd_trace.Option{dd_http.RTWithResourceNamer(resourceNamer)}, opts...)
```

File: client/options.go (hunk 1 above)
Comment (lonnblad) [reply client/options.go:44]: maybe add a line in the description of the function, that when using this in a lambda, make sure to set the options and then an example of how to set it

Comment (stroem) [reply client/options.go:49]: Updated.

**Review by Ekberg — APPROVED** [review]

**Pull request merged**